File: Structural/Facade/Cpp/SoundSystem.cpp
// SoundSystem.cpp

#include "SoundSystem.h"

void SoundSystem::turnOn() {
  std::cout << "Sound System is ON." << std::endl;
}

void SoundSystem::turnOff() {
  std::cout << "Sound System is OFF." << std::endl;
}

void SoundSystem::setVolume(int level) {
  std::cout << "Sound System volume set to: " << level << std::endl;
}

File: Structural/Facade/JavaScript/soundSystem.js
// soundSystem.js

class SoundSystem {
  turnOn() {
    console.log("Sound System is ON.");
  }

  turnOff() {
    console.log("Sound System is OFF.");
  }

  setVolume(level) {
    console.log(`Sound System volume set to: ${level}`);
  }
}

module.exports = SoundSystem;
